src/Aspire.Hosting.Yarp/YarpServiceExtensions.cs

```diff
@@ -85,7 +69,19 @@ public static IResourceBuilder<YarpResource> AddYarp(
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
     public static IResourceBuilder<YarpResource> WithConfigFile(this IResourceBuilder<YarpResource> builder, string configFilePath)
     {
-        builder.Resource.ConfigFilePath = configFilePath;
+        builder.Resource.ConfigurationBuilder.WithConfigFile(configFilePath);
+        return builder;
+    }
+
+    /// <summary>
+    /// Configure the YARP resource.
+    /// </summary>
+    /// <param name="builder">The YARP resource to configure.</param>
+    /// <param name="configurationBuilder">The delegate to configure YARP.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<YarpResource> Configure(this IResourceBuilder<YarpResource> builder, Action<IYarpConfigurationBuilder> configurationBuilder)
```

Comment:
Pattern is usually WithX. So `WithConfiguration`:

```suggestion
    public static IResourceBuilder<YarpResource> WithConfiguration(this IResourceBuilder<YarpResource> builder, Action<IYarpConfigurationBuilder> configurationBuilder)
```